bpf/go_nethttp.c

```diff
@@ -117,13 +127,16 @@ int uprobe_readRequestReturns(struct pt_regs *ctx) {
     return 0;
 }
 
-static __always_inline int writeHeaderHelper(struct pt_regs *ctx, u64 req_offset) {
-        bpf_dbg_printk("=== uprobe/WriteHeader === ");
+SEC("uprobe/ServeHTTP_ret")
```

Comment:
We used to have a return probe on `handler.ServeHTTP`. This caused issues, because chained handlers would appear as if running multiple requests. 

We fixed this by making the return probe be on WriteHeader, so the first call to WriteHeader would send the request back and subsequent calls wouldn't.

However, we later discovered that we can further simplify by attaching a probe on `serverHandler.ServeHTTP`. This function is only called once, but we never stopped using WriteHeader as a method of ending the request.

This change fixes that, we now have a return probe on `serverHandler.ServeHTTP` and we no longer use `WriteHeader`. We only probe on the http2 write header version, so that we can tell we have an HTTP2 request and use different offsets to pull the status and the request from the response writer.

We won't need the probe to the http2 write header, if we can do type checks on the ResponseWriter.